pkg/operator/ceph/cluster/controller.go

```diff
@@ -659,17 +667,19 @@ func (c *ClusterController) removeFinalizer(obj interface{}) {
 	logger.Warningf("giving up from removing the %s cluster finalizer", fname)
 }
 
-func (c *ClusterController) updateClusterStatus(namespace, name string, state cephv1.ClusterState, message string) error {
+func (c *ClusterController) updateClusterStatus(cluster *cluster, name string, state cephv1.ClusterState, message string) error {
 	// get the most recent cluster CRD object
-	cluster, err := c.context.RookClientset.CephV1().CephClusters(namespace).Get(name, metav1.GetOptions{})
+	clusterObj, err := c.context.RookClientset.CephV1().CephClusters(cluster.Namespace).Get(name, metav1.GetOptions{})
 	if err != nil {
-		return fmt.Errorf("failed to get cluster from namespace %s prior to updating its status: %+v", namespace, err)
+		return fmt.Errorf("failed to get cluster from namespace %s prior to updating its status: %+v", cluster.Namespace, err)
 	}
 
-	// update the status on the retrieved cluster object
-	cluster.Status = cephv1.ClusterStatus{State: state, Message: message}
-	if _, err := c.context.RookClientset.CephV1().CephClusters(cluster.Namespace).Update(cluster); err != nil {
-		return fmt.Errorf("failed to update cluster %s status: %+v", cluster.Namespace, err)
+	cluster.status.LastUpdateTime = time.Now().Format(time.RFC3339)
+	cluster.status.SetState(state)
+	clusterObj.Status = cluster.status
```

Comment:
Rather than assigning the overall `cluster.status`, could you just assign each status property directly to `clusterObj.Status`? This will work better with the status coming in #2938.

Reply:
hmm, how about we call `updateClusterStatus` function here as well: https://github.com/rook/rook/pull/2938/files#diff-080e8033099838332e888850718dc67bR106

that way, we'll have a single function to update the `status` object

Reply:
The challenge is that this method wants to update the cluster status (but not the ceph status), and the other method wants to update the ceph status (but not the cluster status). The status that they don't want to update should remain unchanged. How can we accomplish that behavior by calling `updateClusterStatus()` in the other method?

Reply:
`CephStatus` is [a part of](https://github.com/rook/rook/pull/2938/files#diff-080e8033099838332e888850718dc67bR106) `ClusterStatus`, so when we call `updateClusterStatus`, it will persist the entire `cluster.status` to etcd. This should be fine shouldn't it?

Reply:
`updateClusterStatus()` doesn't update the ceph status, right? So it will just set the ceph status to blank if it uses that struct without querying for the ceph status. Rather than set it to blank, we want it to preserve the setting that is already there.

Reply:
ah, I get what you are saying. I was just wondering if it's possible to have only 1 function to update the `CephStatus` object. 
Let me see if I can refactor this and you can review it :+1: